extensions/2.0/Khronos/KHR_texture_basisu/README.md

```diff
@@ -0,0 +1,173 @@
+
+```json
+{
+    "asset": {
+        "version": "2.0"
+    },
+    "extensionsUsed": [
+        "KHR_texture_basisu"
+    ],
+    "extensionsRequired": [
+        "KHR_texture_basisu"
+    ],
+    "textures": [
+        {
+            "extensions": {
+                "KHR_texture_basisu": {
+                    "source": 0
+                }
+            }
+        }
+    ],
+    "images": [
+        {
+            "uri": "image.ktx2"
+        }
+    ]
+}
+```
+
+### JSON Schema
+
+[texture.KHR_texture_basisu.schema.json](schema/texture.KHR_texture_basisu.schema.json)
+
+## KTX v2 Images with Basis Universal Supercompression
+
+To cover a broad range of use cases, this extension allows two Basis Universal formats: ETC1S with BasisLZ and UASTC with optional Zstandard compression.
+
+- KTX header fields for ETC1S with BasisLZ payloads
+  - `supercompressionScheme` must be `1` (BasisLZ).
+  - DFD `colorModel` must be `KHR_DF_MODEL_ETC1S`.
+  - RGB textures must use a single DFD channel with value `KHR_DF_CHANNEL_ETC1S_RGB`.
+  - RGBA textures must use two DFD channels with values `KHR_DF_CHANNEL_ETC1S_RGB` and `KHR_DF_CHANNEL_ETC1S_AAA`.
+
+- KTX header fields for UASTC payloads
+  - `supercompressionScheme` must be `0` (None) or `2` (Zstandard).
+  - DFD `colorModel` must be `KHR_DF_MODEL_UASTC`.
+  - RGB textures must use a single DFD channel with value `KHR_DF_CHANNEL_UASTC_RGB`.
+  - RGBA textures must use a single DFD channel with value `KHR_DF_CHANNEL_UASTC_RGBA`.
+
+Regardless of the format used, these additional restrictions apply for compatibility reasons:
+
+- Swizzling metadata (`KTXswizzle`) must be `rgba` or omitted.
+- Orientation metadata (`KTXorientation`) must be `rd` or omitted.
+- Colorspace information in the DFD must match the expected usage, namely:
+  - For textures with color data, `colorPrimaries` must be set to `KHR_DF_PRIMARIES_BT709`; `transferFunction` to `KHR_DF_TRANSFER_SRGB`.
+  - For textures with non-color data, `colorPrimaries` must be set to `KHR_DF_PRIMARIES_UNSPECIFIED`; `transferFunction` to `KHR_DF_TRANSFER_LINEAR`.
+- `KHR_DF_FLAG_ALPHA_PREMULTIPLIED` flag must not be set unless the material's specification requires premultiplied alpha.
+- When a texture refers to a sampler with mipmap minification or when the sampler is undefined, the KTX image should contain a full mip pyramid.
+  > **Implementation Note**: Depending on the platform features available, engines may need to decompress the texture and generate missing mipmaps at runtime when a full mip pyramid is not available.
+- `pixelWidth` and `pixelHeight` must be multiples of 4.
+  > **Implementation Note**: Exporters may opt to make the dimensions powers of 2. Combined with all mip levels present this would ensure maximum compatibility.
+
+### Using KTX v2 Images with Basis Universal Supercompression for Material Textures
```

Comment:
>  this extension might someday be referenced by an extension that is not related to material textures?

That was the original design. I'm not so sure now given that this extension became fully standalone (and any new extension could in theory override anything it needs).

I still think that support for swizzling and orientation metadata would heavily reduce portability regardless of texture's usage.

Our current materials don't use premultiplied alpha so I moved it up to the "generic" section (although the reference to material's spec remained - probably should be removed).

Reply:
> I still think that support for swizzling and orientation metadata would heavily reduce portability regardless of texture's usage.

Why? Because some engines might ignore this metadata?

Reply:
> Our current materials don't use premultiplied alpha so I moved it up to the "generic" section ...

Current materials explicitly prohibit premultiplied alpha ("stored texels must not be premultiplied"), so it might be worth including an implementation note that premultiplied alpha is prohibited by core spec materials? That is easy to miss.

> I'm not so sure now given that this extension became fully standalone (and any new extension could in theory override anything it needs).

Good point... maybe I'm overthinking it. I would still move the premultiplied alpha item into the material's section, because it explicitly mentions "unless the material's specification..." the others, I think, are fine.

Reply:
@MarkCallow
From the KTX spec:
> Where possible, the preferred default is to use the logical upper-left corner of the image as the texture origin.

glTF core spec also uses `rd` image orientation for JPEG/PNG textures. I don't think that we should diverge from that. Arbitrary swizzling would require either patching shaders on the fly or adding extra uniforms because API-level swizzling is not universally available,